sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVDataSource.scala

```diff
@@ -230,7 +232,7 @@ object TextInputCSVDataSource extends CSVDataSource {
         CSVDataSource.checkHeader(
```

Comment:
Can we remove `CSVDataSource.checkHeader` and switch to `CSVDataSource.checkHeaderColumnNames`? Looks `CSVDataSource.checkHeader` is an overkill and marks hard to read the code.